code/modules/antagonists/traitor/directives/priority_directive.dm

```diff
@@ -0,0 +1,70 @@
+/// This can only be running once at a time, do not run in parallel
+/datum/priority_directive
+	var/name
+	var/objective_explanation
+	var/details
+	var/end_at
+	var/rejected = FALSE
+	var/tc_reward
+	VAR_PRIVATE/list/teams = list()
+
+/datum/priority_directive/New()
+	. = ..()
+
+/// Check if we are allowed to run this directive or not
+/datum/priority_directive/proc/check(list/uplinks, list/player_minds)
+	SHOULD_NOT_OVERRIDE(TRUE)
+	teams.Cut()
+	allocate_teams(uplinks, player_minds)
+	return !rejected
+
+/// Allocate teams for this directive. Call reject() to reject this directive and
+/// add_antagonist_team to add antagonist teams.
+/datum/priority_directive/proc/allocate_teams(list/uplinks, list/player_minds)
+
+/// Return the reward type and amount
+/datum/priority_directive/proc/generate(list/uplinks, list/player_minds)
+
+/// Get the tracking target of this atom
+/datum/priority_directive/proc/get_track_atom()
+
+/// Check if we have finished early. We always complete after a set time period.
+/datum/priority_directive/proc/is_completed()
+	return FALSE
+
+/datum/priority_directive/proc/finish()
+	SSdirectives.active_directive = null
+
+/// Activate the directive, requires a list of traitor datums and security minsd
+/datum/priority_directive/proc/activate(list/uplinks, list/player_minds)
```

Comment:
There is a balance, the more verbose something is the harder it is to use during coding. I think generic names are fine when they are tied to a specific class, as it is very clear that this proc belongs to a specific class so is going to do something related to that. The issue you mention before comes not from the fact that the generation proc is confusing, its that it is poorly documented and there are many classes which appear to do the same thing. In of itself, the generate in that setting is fine since it clearly shows that the proc generates the map, the same way in my case `generate` will generate the things required for the directive to function.
There might be some confusion between generate and activate as they have similar names and it will be confusing as to which one does what (I will change that), but check, reject, and finish have nothing wrong with them and should not be more verbose. These are class specific procs, not global ones. As long as finish completes the directive when called, and check ensures that the directive is valid in some way than they are fine.
All functions have documentation for the cases where it is not immediately obvious what they do within the context of the class they are assigned to.

Reply:
I've renamed the functions to differentiate between some that were unclear and to add underscores to indicated that certain function are not public, but I have not made the function names more verbose. They should be clear enough within the context of their class to not cause issue.

Reply:
The ultimate reason why I am suggesting a little bit verbose name is...
let's say you search `.start(`

![image](https://github.com/BeeStation/BeeStation-Hornet/assets/87972842/35e47094-3d3f-47e8-984d-62502072fc2e)

now you should find out which one is actually our targeted classes `start()`. If this has been `start_objective()`, it can filter out all non-related search results. This shares the same idea for why I put `cfc_` to colour class names.

Back to `generate()`, if these were named as...
``` dm
/datum/mapGeneratorModule/proc/generate_MGM()
/datum/mapGenerator/proc/generate_MG()
/datum/async_map_generator/proc/generate_AMG(...)
// I know this is fucking ugly but hold on
```
now you can search different generate procs because it won't merged into the search result! then you can track which generate proc is used how. Since these are all the same `generate()`, you're supposed to check which one is which specific datum's generate that you want to follow the logic.

So, even if you know what `generate()` does exactly, you **do not have any ability** to check out full codes for which line is actually the `generate()` you are looking for at a glance. You should filter out the noisy result of procs with the same name. Documentation isn't the issue. You can't sort out the full logic at a glance. **This is the ultimate reason.**



It's kinda related to how you start approaching from between a class or the amalgam of codes.
`class -> proc name` then it makes sense. This is starting from a tree, then going to the forest view. 
`/datum/mapGenerator/proc/generate()`. Yes, We know class. You get the info about `/datum/mapGenerator`, and it does its `generate()`, it's obvious.

but the truth is, if people don't know about a tree you know, people actually see a forest(=search), then try to understand which specific tree is located in the forest - `proc name -> class`
* `█████████████/proc/generate()`: Can you guess what this does?
* `█████████████████████/proc/generate()`: There's another one. What's this? It looks the same!

This is what they would see. Of course, you can go closer to this proc, and check which class that is, but you can't do that every tree. It takes your time to read, and it's already a cost. Especially when your intent was to understand the full usage of the specific class's proc...
Since you're seeing things from the forest view, you can't figure out which tree is specific one you're looking for.
* `█████████████/proc/generate_MG()`: same generator, but you can guess it'
* `█████████████████████/proc/generate_AMG()`. 

I know this is something dick, but I believe we should do this because it's open source, and new coders should be able to follow the code quickly.

Reply:
If people want to search for references of a function, then they can use the reference search feature provided by the VS code extension, or search for the typepath first.

Functions do not need to specify what they do independant of their class, since they are not independant of their class. Adding acronyms to the end of function names makes them look worse, makes them harder to type and doesn't help readability. generate_MG and generate_AMG, now have 2 things at the end of them that have absolutely no meaning.

The only case where additions to function names should be used is where they are standardised.
For example, in C#, async functions typically are extended with "async".
GetFile()
GetFileAsync()

> █████████████/proc/generate(): Can you guess what this does?
    █████████████████████/proc/generate(): There's another one. What's this? It looks the same!

Many OOP languages enforce that methods exist on types. Many C# methods would make no sense if you removed the class that they are attached to.
File.Read() is very different to Console.Read(), but you know what they are doing based on context of the class.

Reply:
> # they can use the reference search feature provided by the VS code extension

💀 I've been living in a meaningless pain
Now I see how it was managed.